src/libraries/System.Private.CoreLib/src/System/DateOnly.cs

```diff
@@ -0,0 +1,846 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Globalization;
+
+namespace System
+{
+        internal enum ParseOperationResult
+        {
+            Success,
+            WrongStyles,
+            ParseFailure,
+            WrongParts,
+            BadFormatSpecifier
+        }
```

Comment:
We can't share the same enum that's used with DateTime?

Also, indentation is off here.  And if we can't share it, it should probably be a nested enum under DateOnly rather than being System.ParseOperationResult.

Reply:
I tried to avoid sharing that with DateTimeParse.ParseFailureKind enum because they include different things specific to DateTime parsing and I wanted to avoid merging more values there that are not related to DateTime. I didn't make this enum nested inside DateOnly because I used it in TimeOnly too. we can still move it and reference it with the DateOnly but I thought this will be ugly. 

For the rest of the comments, I'll try to address this in other PR later.

Reply:
> I wanted to avoid merging more values there that are not related to DateTime

I think it's fine to add more.  DateTime, DateTimeOffset, DateOnly, and TimeOnly are all logically related and part of the same system, sharing many similar errors, share format and styles, etc.